ts/store/reducers/entities/messages/allPaginated.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * A reducer to store all messages with pagination
+ */
+
+import * as pot from "italia-ts-commons/lib/pot";
+import { getType } from "typesafe-actions";
+
+import {
+  loadNextPageMessages,
+  loadPreviousPageMessages,
+  reloadAllMessages
+} from "../../../actions/messages";
+import { clearCache } from "../../../actions/profile";
+import { Action } from "../../../actions/types";
+import { GlobalState } from "../../types";
+import { UIMessage } from "./types";
+
+export type Cursor = string;
+
+// An array of messages id
+export type AllPaginated = pot.Pot<
+  { page: ReadonlyArray<UIMessage>; previous?: Cursor; next?: Cursor },
+  string
+>;
+
+const INITIAL_STATE: AllPaginated = pot.none;
+
+const reducer = (
+  state: AllPaginated = INITIAL_STATE,
+  action: Action
+): AllPaginated => {
+  switch (action.type) {
+    case getType(reloadAllMessages.request):
+    case getType(reloadAllMessages.success):
+    case getType(reloadAllMessages.failure):
+      return reduceReloadAll(state, action);
+
+    case getType(loadNextPageMessages.request):
+    case getType(loadNextPageMessages.success):
+    case getType(loadNextPageMessages.failure):
+      return reduceLoadNextPage(state, action);
+
+    case getType(loadPreviousPageMessages.request):
+    case getType(loadPreviousPageMessages.success):
+    case getType(loadPreviousPageMessages.failure):
+      return reduceLoadPreviousPage(state, action);
+
+    case getType(clearCache):
+      return INITIAL_STATE;
+
+    default:
+      return state;
+  }
+};
+
+const reduceReloadAll = (
+  state: AllPaginated = INITIAL_STATE,
+  action: Action
+): AllPaginated => {
+  switch (action.type) {
+    case getType(reloadAllMessages.request):
+      return pot.toLoading(state);
+
+    case getType(reloadAllMessages.success):
+      return pot.some({
+        page: action.payload.messages,
+        previous: action.payload.pagination.previous,
+        next: action.payload.pagination.next
+      });
+
+    case getType(reloadAllMessages.failure):
+      return pot.toError(state, action.payload.message);
+
+    default:
+      return state;
+  }
+};
+
+const reduceLoadNextPage = (
+  state: AllPaginated = INITIAL_STATE,
+  action: Action
+): AllPaginated => {
+  switch (action.type) {
+    case getType(loadNextPageMessages.request):
+      return pot.toLoading(state);
+
+    case getType(loadNextPageMessages.success):
+      return pot
+        .toOption(state)
+        .map(previousState =>
+          pot.some({
+            page: previousState.page.concat(action.payload.messages),
+            next: action.payload.pagination.next
+          })
+        )
+        .getOrElse(
+          pot.some({
+            page: [...action.payload.messages],
+            next: action.payload.pagination.next
+          })
+        );
```

Comment:
I kept the same syntax but in a slightly larger scope (now it is part of the state rather than the entire one) https://github.com/pagopa/io-app/pull/3441/files#diff-482523dcf54c84f4e40e09c47ee2eeca62c92423ef82ff98aae91839199d8c16R108
Haven't really thought about it. Please let me know if you think it should be updated 😄